pkg/controller/postgresql.go

```diff
@@ -37,6 +51,11 @@ func (c *Controller) clusterListFunc(options api.ListOptions) (runtime.Object, e
 		return nil, fmt.Errorf("could not extract list of postgresql objects: %v", err)
 	}
 
+	if time.Since(c.lastClusterSyncTime) <= c.opConfig.ResyncPeriod {
```

Comment:
This function can be called concurrently, therefore, I'd protect access to the c.lastClusterSyncTime with a lock.